plugins/modules/web_app_pool_info.yml

```diff
@@ -0,0 +1,286 @@
+---
+# Copyright: (c) 2024, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION:
+  module: web_app_pool_info
+  short_description: Get information on IIS Web Application Pools
+  description:
+    - Returns information about IIS Web Application Pools
+  options:
+    name:
+      description:
+        - Name of the application pool.
+        - When not specified, information of all existing pools will be fetched
+      type: str
+      required: false
+  seealso:
+    - module: microsoft.iis.web_app_pool
+    - module: community.windows.win_iis_virtualdirectory
+    - module: community.windows.win_iis_webapplication
+    - module: community.windows.win_iis_webbinding
+    - module: community.windows.win_iis_website
+  author:
+    - Ron Gershburg (@rgershbu)
+
+EXAMPLES: |
+  - name: Return information about an existing application pool
+    microsoft.iis.web_app_pool_info:
+      name: DefaultAppPool
+    register: stored_info
+  
+  - name: Returns information about all application pools that exist on the system
+    microsoft.iis.web_app_pool_info:
+      name: 
```

Comment:
A better example might be to omit this option altogether rather than the `null` one you have here.

Reply:
Just for clarification, you suggest me to remove the 'name' key in the example or remove this example all together?
I added this example just to specify the user that if he wants to get all the pools he can.

Reply:
Yep make it just

```yaml
  - name: Returns information about all application pools that exist on the system
    microsoft.iis.web_app_pool_info:
    register: stored_info_all
```

Reply:
https://github.com/ansible-collections/microsoft.iis/commit/49e95c9684ce9ce31dba2a03438eda006795c927